scvi/external/cell2location/_model.py

```diff
@@ -0,0 +1,572 @@
+        :param n_samples: number of posterior samples to generate (1000 is recommended, reduce if you get GPU memory error)
+        :param save_samples: save samples in addition to sample mean, 5% quantile, SD.
+        :param return_samples: return summarised samples in addition to saving them in `self.samples`
+        :param mean_field_slot: string, which mean_field slot to sample? 'init_1' by default
+        :return: nothing, a dictionary of dictionaries (mean, 5% quantile, SD, optionally all samples) with numpy arrays for each variables is added to self.adata.uns['mod'].
+        Optional dictionary of all samples contains parameters as numpy arrays of shape ``(n_samples, ...)``
+        """
+
+        self.n_samples = n_samples
+        self.n_sampl_batches = int(np.ceil(n_samples / num_samples_batch))
+        self.num_samples_batch = num_samples_batch
+
+        if (node == 'all'):
+            # Sample all parameters - might use a lot of GPU memory
+
+            self.sample_all(self.n_sampl_batches, num_samples_batch=self.num_samples_batch)
+
+            self.param_names = list(self.adata.uns['mod']['post_samples'].keys())
+
+            self.adata.uns['mod']['post_sample_means'] = {v: self.adata.uns['mod']['post_samples'][v].mean(axis=0)
+                                                 for v in self.param_names}
+            self.adata.uns['mod']['post_sample_q05'] = {v: np.quantile(self.adata.uns['mod']['post_samples'][v], 0.05, axis=0)
+                                               for v in self.param_names}
+            self.adata.uns['mod']['post_sample_q95'] = {v: np.quantile(self.adata.uns['mod']['post_samples'][v], 0.95, axis=0)
+                                               for v in self.param_names}
+            self.adata.uns['mod']['post_sample_sds'] = {v: self.adata.uns['mod']['post_samples'][v].std(axis=0)
+                                               for v in self.param_names}
+
+            if not save_samples:
+                del self.adata.uns['mod']['post_samples']
+
+        else:
+            self.sample_node(node, self.n_sampl_batches,
+                             batch_size=self.num_samples_batch, suff='')
+    
+    @property
+    def _plan_class(self):
+        return PyroTrainingPlan
+
+    @property
+    def _data_loader_cls(self):
+        return AnnDataLoader
+    
+    @staticmethod
+    def align_plot_stability(fac1, fac2, name1, name2,
+                             align=True, return_aligned=False):
+        r"""Align columns between two np.ndarrays using scipy.optimize.linear_sum_assignment,
+            then plot correlations between columns in fac1 and fac2, ordering fac2 according to alignment
+
+        :param fac1: np.ndarray 1, factors in columns
+        :param fac2: np.ndarray 2, factors in columns
+        :param name1: axis x name
+        :param name2: axis y name
+        :param align: boolean, match columns in fac1 and fac2 using linear_sum_assignment?
```

Comment:
we use numpydoc style docstrings. They are easier to read and write.

Reply:
Yes, I need to change that.